File: qc-core/src/main/java/com/eager/questioncloud/core/domain/subscribe/infrastructure/SubscribeEntity.java
package com.eager.questioncloud.core.domain.subscribe.infrastructure;

import com.eager.questioncloud.core.domain.subscribe.model.Subscribe;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import java.time.LocalDateTime;
import lombok.AccessLevel;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Getter
@Entity
@Table(name = "subscribe")
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class SubscribeEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column
    private Long subscriberId;

    @Column
    private Long creatorId;

    @Column
    private LocalDateTime createdAt;

    @Builder
    public SubscribeEntity(Long id, Long subscriberId, Long creatorId, LocalDateTime createdAt) {
        this.id = id;
        this.subscriberId = subscriberId;
        this.creatorId = creatorId;
        this.createdAt = createdAt;
    }

    public Subscribe toModel() {
        return Subscribe.builder()
            .id(id)
            .subscriberId(subscriberId)
            .creatorId(creatorId)
            .createdAt(createdAt)
            .build();
    }

    public static SubscribeEntity from(Subscribe subscribe) {
        return SubscribeEntity.builder()
            .id(subscribe.getId())
            .subscriberId(subscribe.getSubscriberId())
            .creatorId(subscribe.getCreatorId())
            .createdAt(subscribe.getCreatedAt())
            .build();
    }
}

File: qc-infra/qc-rdb/src/main/kotlin/com/eager/questioncloud/subscribe/entity/SubscribeEntity.kt
package com.eager.questioncloud.subscribe.entity

import com.eager.questioncloud.subscribe.domain.Subscribe
import jakarta.persistence.*
import java.time.LocalDateTime

@Entity
@Table(name = "subscribe")
class SubscribeEntity private constructor(
    @GeneratedValue(strategy = GenerationType.IDENTITY) @Id var id: Long = 0,
    @Column var subscriberId: Long,
    @Column var creatorId: Long,
    @Column var createdAt: LocalDateTime
) {
    fun toModel(): Subscribe {
        return Subscribe(id, subscriberId, creatorId, createdAt)
    }
    
    companion object {
        fun from(subscribe: Subscribe): SubscribeEntity {
            return SubscribeEntity(
                subscribe.id,
                subscribe.subscriberId,
                subscribe.creatorId,
                subscribe.createdAt
            )
        }
    }
}
